kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWriteSuiteBase.scala

```diff
@@ -316,8 +328,51 @@ trait DeltaTableWriteSuiteBase extends AnyFunSuite with TestUtils {
       stageData(txnState, partValues, partData)
     }
 
-    val combineActions = inMemoryIterable(actions.reduceLeft(_ combine _))
-    commitTransaction(txn, engine, combineActions)
+    actions.reduceLeftOption(_ combine _) match {
+      case Some(combinedActions) =>
+        val combineActions = inMemoryIterable(combinedActions)
+        commitTransaction(txn, engine, combineActions)
+      case None =>
+        commitTransaction(txn, engine, emptyIterable[Row])
+    }
+  }
+
+  /** Utility to create table, with no data */
+  def createTable(
+      engine: Engine = defaultEngine,
+      tablePath: String,
+      schema: StructType,
+      partCols: Seq[String] = Seq.empty,
+      clock: Clock = () => System.currentTimeMillis,
+      tableProperties: Map[String, String] = null): TransactionCommitResult = {
+
+    appendData(
+      engine,
+      tablePath,
+      isNewTable = true,
+      schema,
+      partCols,
+      data = Seq.empty,
+      clock,
+      tableProperties)
+  }
+
+  /** Update an existing table - metadata only changes (no data changes) */
+  def updateTable(
```

Comment:
nit - rename to something that reflects the docs/usage `updateTableMetadata`? `commitMetadataChange`?